braid/_braid.c

```diff
@@ -2519,20 +2529,103 @@ _braid_FRefineSpace(braid_Core   core,
    {
       *refined_ptr = 0;
    }
-
+    
    /* Reset r_space */
    _braid_CoreElt(core, r_space) = 0;
    return _braid_error_flag;
 }
 
+/*----------------------------------------------------------------------------
+ * Helper routine for periodic problems.  Compute the divisor by which the
+ * number of points must be divisible and also the number of extra needed.
+ *----------------------------------------------------------------------------*/
+
+braid_Int
+_braid_PeriodicCheckNumPoints(braid_Core   core,
+                              braid_Int    npoints,
+                              braid_Int   *divisor_ptr,
+                              braid_Int   *nextra_ptr)
+{
+   braid_Int  divisor, nextra, level, size, cfactor;
+
+   /* Compute coarsening factor divisor */
+   divisor = 1;
+   nextra  = 0;
+   size = npoints;
+   for (level = 0; level < (_braid_CoreElt(core, max_levels) - 1); level++)
+   {
+      _braid_GetCFactor(core, level, &cfactor);
+      size    /= cfactor;
+      divisor *= cfactor;
+      if (size < cfactor)
+      {
+         break;
+      }
+   }
+   if (divisor < npoints)
+   {
+      divisor *= cfactor;
+   }
+   nextra = divisor - npoints;
```

Comment:
Let's say that npoints = 11, max_levels=2, and cfactor = 2 everywhere.  Then
nextra = divisor - npoints = 2 - 11 = 9.
Right?  That will give us 20 points, and carry about a near uniform refinement.
This is the logic that we're going for, right?  We could alternatively add just 1 point in time, for this case, but that raises the question of where you should add this point.

Reply:
Actually, you'll get 4-11 = -7, which is messed up.  The logic works if the loop breaks out due to the if statement, but apparently it's not correct when the loop exits because of max_levels.  Let me work on it.  Good catch!

Reply:
I'm pretty sure I have the logic right now.